test/newNameLength.test.ts

```diff
@@ -26,10 +27,10 @@ const longName = 'eks-blueprint-with-an-extra-kind-of-very-long-name-that-is-def
 
 test("Testing ID length error!", () => {
   expect(() => getConstraintsDataSet().at(1)?.build(app, longName))
-    .toThrow!('Managed Node Groups ID must be no more than 63 characters long!');
+    .toThrow(z.ZodIssueCode.too_big);
```

Comment:
We need to see the actual field in the error message. Let's update the test in a way that will include validation of the field. You can pass RegExp to the toThrow() to match the string.

Reply:
Ya this was the last thing I was trying to figure out yesterday. I was trying to find a way to extract the "Message" parameter from the ZodError but was not having luck with that. I did find .too_big worked to a degree but not what I am looking for. 
Like how below the custom error messages I have. If you see ZodErrors its within there but not having luck extracting it.
`z.string() .min(this.min ?? 0, { message: "Managed Node Groups " + key + " must be no less than 1 characters long!" })
`
I figured if I get that to work I can easily have the errors send out without the need for try-catch blocking each section to throw the errors.

And I am not quite sure how RegExp can help here? I have a little Regex experience from Java which from a google RegExp is another way to say it. But Regexp passed into toThrow to match the string? Doesn't toThrow simply check if the error thrown matches the given string? Why would you want to match inside of that?